### PR TITLE
nest-cli: repackage using buildNpmPackage

### PR DESCRIPTION
**Pull request by ehllie**
###### Description of changes
Repackages `nodePackages.@nestjs/cli` to be built using buildNpmPackage as per the suggestion in #229475

###### Things done
Moved `nest-cli` to `pkgs/development/tools` and removed it from `node-packages.json`.
Had to regenerate `package-lock.json` after running `npm upgrade` to fix the build failing on Darwin.
Once #229639 gets merged I will also include it in `aliases.nix` that it adds.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5438,6 +5438,8 @@ with pkgs;
 
   n2n = callPackage ../tools/networking/n2n { };
 
+  nest-cli = callPackage ../development/tools/admin/nest-cli { };
```
Comment (lilyinstarlight) [inline pkgs/top-level/all-packages.nix:5441]: The directory was put in `pkgs/development/tools/nest-cli`, not `pkgs/development/tools/admin/nest-cli`. Which one were you intending?

Comment (ehllie) [reply pkgs/top-level/all-packages.nix:5441]: I had meant the directory I put it in, just got confused when including it in `all-packages.nix` 😄

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please rebase and add an alias

**Comment by ehllie** [comment]
> Please rebase and add an alias

I have done that, but I have an issue I pointed out here https://github.com/NixOS/nixpkgs/pull/229639#discussion_r1186458746. As it stands `nodePackages.@nestjs/cli` does not evaluate because it shadows the package in `node-packages.nix`, which I would like to avoid having to manually edit or re generate.

**Comment by ehllie** [comment]
I've also deleted `@nestjs/cli` from `node-packages.nix` so that the alias is not an error in case we want to stay with the current iteration of the alias file

**Comment by dotlambda** [comment]
> Had to regenerate `package-lock.json` after running `npm upgrade` to fix the build failing on Darwin.

Did you ask upstream to regenerate their lock file? We cannot add 15k lines of code for every single package.

**Pull request merged**